hier_config/__init__.py

```diff
@@ -52,15 +52,24 @@ class HConfig(HConfigChild):
 
     """
 
-    def __init__(self, hostname, os, options):
+    def __init__(self, hostname=None, os=None, options=None, host=None):
         self._hostname = hostname
         self.os = os
-        self.options = dict(options)
+        self.options = options
         self._text = str()
         self._logs = list()
         self.children = []
         self.children_dict = {}
 
+        if all((self.hostname, self.os, self.options)) is not None:
+            from hier_config.host import Host
+            self.host = Host(self.hostname, self.os, self.options)
+        elif host is not None:
+            assert hasattr(host, (hostname, os, options))
+            self.host = host
```

Comment:
If the `HConfig` class is called like `HConfig(Host)` then `self._hostname` and `self.os` will be `None`. Consider changing all consumption of these properties to `self.host.hostname/os`.